src/Microsoft.AspNetCore.Server.Kestrel.Https/HttpsConnectionAdapter.cs

```diff
@@ -70,7 +70,7 @@ public HttpsConnectionAdapter(HttpsConnectionAdapterOptions options, ILoggerFact
                             }
                         }
 
-                        var certificate2 = ConvertToX509Certificate2(certificate);
+                        var certificate2 = (X509Certificate2)certificate;
                         if (certificate2 == null)
```

Comment:
This check should be unnecessary unless the previous line is converted to a safe cast.